cv32/bsp/handlers.S

```diff
@@ -23,13 +23,110 @@
 .global __no_irq_handler
 .global sw_irq_handler
 .global verification_irq_handler
+.global software_irq_handler
```

Comment:
What is the difference between `software_irq_handler` and the existing `sw_irq_handler`?

Reply:
software_irq_handler specifically maps to MSIP (mip[3] in vectored interrupt mode) versus the sw_irq_handler at mip[0].  The sw_irq_handler at mip[0] is still needed for handling basic exception in vectored mode.   

I would be open to a better name for the mip[3] handler.

Reply:
Excellent idea Craig, I will incorporate this into the change.  I will also document in the markdown.